libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DNS.c

```diff
@@ -1112,321 +1109,343 @@ uint16_t usType = 0U;
 	/* Ensure that the buffer is of at least minimal DNS message length. */
 	if( uxBufferLength < sizeof( DNSMessage_t ) )
 	{
-		return dnsPARSE_ERROR;
+		xReturn = pdFALSE;
 	}
```

Comment:
I would suggest to begin an else clause here, like :
~~~c
    if( uxBufferLength < sizeof( DNSMessage_t ) )
    {
        xReturn = pdFALSE;
    }
    else
    {
        uxSourceBytesRemaining = uxBufferLength;

        /* Parse the DNS message header. Map the byte stream onto a structure 
        * for easier access. */
        pxDNSMessageHeader = ipPOINTER_CAST( DNSMessage_t *, pucUDPPayloadBuffer );

        /* Introduce a do {} while (0) to allow the use of breaks. */
        do
        {
        size_t uxBytesRead = 0U;
        // etc
~~~

Reply:
Good catch. thanks!